examples/tests/Earthfile

```diff
@@ -716,6 +712,79 @@ true-false-flag-invalid:
     DO +RUN_EARTHLY --earthfile=true-false-flag-invalid.earth --should_fail=true --target=+run-maybe --post_command="2>output3.txt"
     RUN cat output3.txt | grep 'invalid argument for flag .*--no-cache.*expected bool'
 
+scrub-git-clone-credentials:
+    # This token is setup with read only access to public repos
+    ARG PUBLIC_TOKEN="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"
```

Comment:
Something feels wrong about this... I know the token is limited, but it's possible it also has access to read some user data or something miscellaneous (now or in the future) that the public shouldn't have access to.

Reply:
It might be possible to extend the self-hosted git repo test to work over https auth; then we could ensure the configured password doesn't have access to anything else. I'll take a look to see how doable that is.

Reply:
It was a bit more finessing than expected, but I got a `+git-webserver` target working that allows us to perform this test using a self-hosted, self-signed https webserver.